CI: drop actions-rs, add MSRV/nightly, outdated, miri, & scheduled checks

utkarshgupta137 opened the pull request.

utkarshgupta137 commented:
@lunacookies I'm not a CI guy. Can you check once?

lunacookies commented:
Sorry, I have no idea what that failure is from. The actual GitHub Actions workflow looks right to me. Maybe we can get away with a newer MSRV?

utkarshgupta137 commented:
Ugh, I give up.

lunacookies commented:
:( if you can be bothered you could try using Rust 1.68, just to see if there’s something fundamentally broken about the approach used here, or if instead there’s something about those older Rust versions that is incompatible with etcetera.

utkarshgupta137 commented:
Our actual MSRV should be 1.48 (due to the windows-sys dep of the `home` crate), but the macOS build fails below 1.54. I'm not even sure what change in [1.54](https://github.com/rust-lang/rust/releases/tag/1.54.0) fixed the issue.

Also, `actions-rs` has not been updated in 2+ years, so I've switched to the GitHub action maintained by dtolnay. It also results in faster CI since the latest images come pre-installed with rustup.

I've also copied the `outdated` & `miri` jobs from [anyhow](https://github.com/dtolnay/anyhow/blob/master/.github/workflows/ci.yml).

@lunacookies Review & Merge :)

lunacookies commented on .github/workflows/ci.yaml:
```diff
@@ -1,45 +1,72 @@
+      - master
+    tags:
+      - '*'
+  schedule:
+    - cron: '0 0 * * FRI'
```
Is there a reason for re-running CI every Friday?

utkarshgupta137 replied:
It is to let us know if a new rust version breaks CI or if any of the deps become outdated. It would be best to keep windows-sys crate version & the copied code in sync with the `home` crate. It would be fine to run this only twice or even once a month.

lunacookies replied:
Ah okay, that makes sense.

lunacookies reviewed (approved):
Thanks!
